test/python/scheduler/test_basic_scheduler.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test cases for the pulse scheduler passes."""
+
+from qiskit import QuantumRegister, ClassicalRegister, QuantumCircuit
+from qiskit.scheduler.basic_scheduler import schedule
+
+from qiskit.test.mock import FakeOpenPulse2Q
+from qiskit.test import QiskitTestCase
+
+
+class TestBasicSchedule(QiskitTestCase):
+    """Scheduling tests."""
+
+    def setUp(self):
+        self.backend = FakeOpenPulse2Q()
+
+    def test_forward_pass(self):
+        """Test forward (late) scheduling."""
+        q = QuantumRegister(2)
+        c = ClassicalRegister(2)
+        qc = QuantumCircuit(q, c)
+        qc.x(q[0])
+        qc.x(q[1])
+        qc.barrier(q[1])
+        qc.x(q[1])
+        qc.barrier(q[0], q[1])
+        qc.cx(q[0], q[1])
+        qc.measure(q, c)
+        sched = schedule(qc, self.backend, push_forward=True)
```

Comment:
which direction is "forward"?
I always thought forward is 
this way <------

but here it seems like it is 
----------> this way!

Reply:
Forward, in this case, means to **push forward to a later time**. This is why I advocate for a more explicit nomenclature in an earlier comment.